### PR TITLE
Detection return empty result #347

### PR DESCRIPTION
**Pull request by amper43**
fixes #348

### CODE REVIEW

File: analytics/analytics/analytic_unit_manager.py [hunk 1]
```diff
@@ -16,9 +16,9 @@
 AnalyticUnitId = str
 
 
-def get_detector_by_type(detector_type: str, analytic_unit_type: str) -> detectors.Detector:
+def get_detector_by_type(detector_type: str, analytic_unit_type: str, analytic_unit_id: str) -> detectors.Detector:
```
Comment (jonyrock) [inline analytics/analytics/analytic_unit_manager.py]: Please use type definition here: https://github.com/hastic/hastic-server/blob/9b6f5482bc53bdd9d6cf7615e8940914d8ed81a6/analytics/analytics/analytic_unit_manager.py#L16

File: analytics/analytics/detectors/pattern_detector.py [hunk 2]
```diff
@@ -32,11 +32,13 @@ def resolve_model_by_pattern(pattern: str) -> models.Model:
 
 class PatternDetector(Detector):
 
-    def __init__(self, pattern_type):
+    def __init__(self, pattern_type, analytic_unit_id):
```
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]: add type declaration here plz

File: analytics/analytics/detectors/pattern_detector.py [hunk 3]
```diff
@@ -63,14 +66,21 @@ def detect(self, dataframe: pd.DataFrame, cache: Optional[models.ModelCache]) ->
 
     def recieve_data(self, data: pd.DataFrame, cache: Optional[ModelCache]) -> Optional[dict]:
         self.bucket.receive_data(data.dropna())
-        if cache != None:
-            self.window_size = cache['WINDOW_SIZE']
+        #if cache != None:
```
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]: can we remove it?

File: analytics/analytics/detectors/pattern_detector.py [hunk 4]
```diff
@@ -29,10 +29,10 @@ def resolve_model_by_pattern(pattern: str) -> models.Model:
         return models.CustomModel()
     raise ValueError('Unknown pattern "%s"' % pattern)
 
-
+AnalyticUnitId = str
```
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]: please import it like there: https://github.com/hastic/hastic-server/blob/d4f6783652431b32210114ae730ddbf38f00dba4/server/src/models/analytics_task_model.ts#L1

File: analytics/analytics/detectors/pattern_detector.py [hunk 5]
```diff
@@ -29,14 +29,16 @@ def resolve_model_by_pattern(pattern: str) -> models.Model:
...
```
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]: can you import this definition from analytic_unit_manager.py ?

Comment (amper43) [reply analytics/analytics/detectors/pattern_detector.py]: @jonyrock  it makes cyclic link

**Pull request merged**